cmd/kola/testiso.go

```diff
@@ -0,0 +1,327 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/coreos/mantle/util"
+	"github.com/pkg/errors"
+
+	"github.com/coreos/mantle/sdk"
+
+	ignconverter "github.com/coreos/ign-converter"
+	ignv3types "github.com/coreos/ignition/v2/config/v3_0/types"
+	"github.com/spf13/cobra"
+
+	"github.com/coreos/mantle/kola"
+	"github.com/coreos/mantle/platform"
+)
+
+var (
+	cmdTestIso = &cobra.Command{
+		Run:    runTestIso,
+		PreRun: preRun,
+		Use:    "testiso",
+		Short:  "Test a CoreOS PXE boot or ISO install path",
+	}
+	// TODO expose this as an API that can be used by cosa too
+	consoleKernelArgument = map[string]string{
+		"amd64-usr":   "ttyS0",
+		"ppc64le-usr": "hvc0",
+		"arm64-usr":   "ttyAMA0",
+		"s390x-usr":   "ttysclp0",
+	}
+)
+
+func init() {
+	root.AddCommand(cmdTestIso)
+}
+
+func runTestIso(cmd *cobra.Command, args []string) {
+	if err := doTestIso(cmd, args); err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		os.Exit(1)
+	}
+}
+
+func doTestIso(cmd *cobra.Command, args []string) error {
+	//var err error
+
+	if kola.CosaBuild == nil {
+		return fmt.Errorf("Must provide --cosa-build")
+	}
+
+	if kola.CosaBuild.BuildArtifacts.Kernel.Path != "" {
+		return errors.Wrapf(testLegacyInstaller(), "testing legacy installer")
+	} else {
+		return fmt.Errorf("build %s has no installer kernel", kola.CosaBuild.Name)
+	}
+}
+
+func testLegacyInstaller() error {
+	builddir := filepath.Dir(kola.Options.CosaBuild)
+	metalimg := kola.CosaBuild.BuildArtifacts.Metal.Path
+
+	builder := platform.NewBuilder(kola.QEMUOptions.Board, "")
+	defer builder.Close()
+	builder.Firmware = kola.QEMUOptions.Firmware
+	builder.AddDisk(&platform.Disk{
+		Size: "12G", // Arbitrary
+	})
+	// Arbitrary
+	builder.Memory = 1536
+
+	builder.InheritConsole = true
+
+	tftpipaddr := "192.168.76.2"
+	var boottype, networkdevice, bootindex, pxeimagepath string
+	switch kola.QEMUOptions.Board {
+	case "amd64-usr":
+		boottype = "pxe"
+		networkdevice = "e1000"
+		pxeimagepath = "/usr/share/syslinux/"
+		break
+	case "ppc64le-usr":
+		boottype = "grub"
+		networkdevice = "virtio-net-pci"
+		break
+	case "s390x-usr":
+		boottype = "pxe"
+		networkdevice = "virtio-net-ccw"
+		tftpipaddr = "10.0.2.2"
+		bootindex = "1"
+	default:
+		return fmt.Errorf("Unsupported arch %s", kola.QEMUOptions.Board)
```

Comment:
`arm64` is in the `consoleKernelArgument` map but would just error out. Should that just be dropped?

Reply:
I kept it for completeness when copying the code from cosa.  I plan to de-dup that code soon, and we want to support arm64 anyways in the future right?

Reply:
right...i did look into arm...should be a matter of pointing to the correct boot images so definitely can be supported.. when  i get some time i'll do that